rfcs/0000-standardize_the_kuadrant_spec.md

```diff
@@ -0,0 +1,238 @@
+
+Given that the lists are explicit the following configuration would only create service monitors for the kuadrant-operator.
+
+```yaml
+spec:
+  service_monitors:
+    configure: true
+    enable: ["kuadrant-operator"]
+```
+If the user was wanting to extend which components had the feature applied to the configuration would look like:
+```yaml
+spec:
+  service_monitors:
+    configure: true
+    enable: ["defaults", "kuadrant-operator"]
+```
+This can be expanded more to where the user does not make use of the dns-operator and therefore does not want service monitors to be created for the feature.
+```yaml
+spec:
+  service_monitors:
+    configure: true
+    enable: ["defaults", "kuadrant-operator"]
+    disable: ["dns-operator"]
+```
+
+## Log Level adjustment.
+Being able to config the log level of the components from the global level has been a feature that has being talked about many times.
+There is currently [https://github.com/Kuadrant/architecture/pull/97](https://github.com/Kuadrant/architecture/pull/97) would cover the finer details of the feature implementation.
+However, it does show an issue with this API design.
+The issue can be resolved.
+
+To describe the issue, there are many components and many log levels.
+How does the API design allow some components to be at one log level while others at a different level.
+There are two approaches that can be taken.
+Both have their pros and cons.
+
+The first method is to have sub-fields under the feature field which would describe the configuration for that field.
+```yaml
+spec:
+  logging_level:
+    info:
+      configure: true
+      enable: []
+      disable: []
+    debug:
+      configure: true
+      enable: []
+      disable: []
+```
+The second method is to use a list based approach.
+In this method an extra field is added to the object.
+```yaml
+spec:
+  logging_level:
+    - level: "info"
+      configure: true
+      enable: []
+      disable: []
+    - level: "debug"
+      configure: true
+      enable: []
+      disable: []
+```
+The list based method seems to have more flexibility than the field based method.
+As the level is defined as a string and not a key, if there is a new component with different log levels added, updating to support those levels does not require updating the CRD.
+However, the first approach is more user readable. 
+
+One thing that would be left up to the feature design, and is out of scope of this RFC is how the elements should be prioritized. 
+For example if a component is listed as enable in two sets of configures, which configuration is added.
+In the case of logging I believe it should be the more verbose configuration.
+```yaml
+spec:
+  logging_level:
+    - level: "error"
+      configure: true
+      enable: ["kuadrant-operator"]
+      disable: []
+    - level: "debug"
+      configure: true
+      enable: ["kuadrant-operator"]
+      disable: []
+```
```

Comment:
Thinking about this again. There might be some thing that we can do with CEL to ensure a component is only configured in one listed object.